Classes/Dimension/ContentDimension.php

```diff
@@ -14,160 +10,121 @@
  * source code.
  */
 
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\DimensionSpace\Dimension;
+
+use Neos\ContentRepository\DimensionSpace\Dimension\Exception\ContentDimensionValuesAreMissing;
 use Neos\Utility\Arrays;
 
 /**
  * The content dimension domain model
  */
 final class ContentDimension
 {
-    /**
-     * @var ContentDimensionIdentifier
-     */
-    protected $identifier;
+    public readonly ContentDimensionIdentifier $identifier;
```

Comment:
IMO we should use [Constructor property promotion](https://stitcher.io/blog/new-in-php-8#constructor-property-promotion-rfc) here (like https://3v4l.org/ECIar#v8.1.1)

Reply:
Yep, same for defaultValue and later on values (see below)